src/Database/PostgreSQL/Consumers/Components.hs

```diff
@@ -345,15 +361,18 @@ spawnDispatcher ConsumerConfig{..} cs cid semaphore
         , "    WHEN id = ANY(" <?> Array1 successes <+> ") THEN " <?> now
         , "    ELSE NULL"
         , "  END"
-        , "WHERE id = ANY(" <?> Array1 (map fst updates) <+> ")"
+        , "WHERE id" <+> operator <+> "ANY (" <?> Array1 (map fst updates) <+> ")"
         ]
       where
+        operator = case ccMode of
+          Standard -> "="
+          Duplicating _field -> "<="
```

Comment:
It would be better to change the type signature of `updateJobs` so that it takes only a single `(idx, Result)` in the deduplicating case :thinking: The problem now is that if something goes awry and multiple ids are passed here, the condition `id <= ANY (...)` will wreak havoc.

Reply:
Also, how were tests passing with this bug? :) They should be updated accordingly.